Formula/sigrok-firmware-fx2lafw.rb

```diff
@@ -0,0 +1,19 @@
+class SigrokFirmwareFx2lafw < Formula
+  desc "Open-source firmware for Cypress FX2 chips"
+  homepage "https://sigrok.org/wiki/Fx2lafw"
+  url "https://sigrok.org/download/source/sigrok-firmware-fx2lafw/sigrok-firmware-fx2lafw-0.1.7.tar.gz"
+  sha256 "a3f440d6a852a46e2c5d199fc1c8e4dacd006bc04e0d5576298ee55d056ace3b"
+
+  depends_on "make" => :build
+  depends_on "sdcc" => :build
+
+  def install
+    system "./configure", "--prefix=#{prefix}"
+    system "make"
+    system "make", "install"
+  end
+
+  test do
+    assert_equal "13", shell_output("ls -1 #{share}/sigrok-firmware/ | grep -v ^l | wc -l").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (`foo --version` or `foo --help`) are not sufficient, as explained in the [formula cookbook](https://docs.brew.sh/Formula-Cookbook).

In most cases, a good test would involve running a simple test case: run `#{bin}/foo input.txt`.
- Then you can check that the output is as expected (with `assert_equal` or `assert_match` on the output of `shell_output`)
- You can also check that an output file was created, if that is expected: `assert_predicate testpath/"output.txt", :exist?`

Some advice for specific cases:
- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks for your feedback. I would respectfully like to suggest that this package is a worthy exception to this rule.

This package is not actually executable/binaries/software. This package is actually assets for the `libsigrok` library. The 'compiled' firmware can only be run on a Cypress FX2 microcontroller. There is no macOS machine that can run these binaries.

The sigrok developers distribute these assets separately to the main `libsigrok` packages. In my humble opinion these assets should be included with the `libsigrok` package, but unfortunately that is not the case.

I submitted a Formula that downloads the sourcecode and compiles the firmware. The sigrok developers also distribute copmiled versions of this firmware, would you rather that I package these?

Fyi I chose the .tar.gz with the sourcecode for two reasons:

1. **Emotional:** Free and open source is awesome and I'd like to support this by tying this package to sourcecode instead of binaries.
2. **Practical:** This provides a bit of additional assurance that the firmware files are exactly what you think they are (based on the source code)

Please let me know what your desired approach is.